File: src/main/java/ch/ethann/tpi/service/TaskService.java
package ch.ethann.tpi.service;

import java.util.List;

import ch.ethann.tpi.excpetion.Task.TaskException;
import ch.ethann.tpi.model.Kanban;
import ch.ethann.tpi.model.Task;

public interface TaskService {
    public List<Task> getTasks(Kanban kanban); 

    public Task getTask(Kanban kanban, String name) throws TaskException;

    public Task createTask(Task task) throws TaskException;

    public void deleteTask(Kanban kanban, String name) throws TaskException;

    public Task updateTask(Task task) throws TaskException;

    public Task renameTask(Kanban kanban, String oldName, String newName) throws TaskException;
}

File: src/main/angular/src/app/@core/service/task.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  constructor(
    private http: HttpClient,
  ) { }

    getTasks(kanbanName: string) {
        return this.http.get('/api/kanban/' + kanbanName + '/task');
    }

    getTask(kanbanName: string, taskName: string) {
        return this.http.get('/api/kanban/' + kanbanName + '/task/' + taskName);
    }

    createTask(kanbanName: string, name: string, assignee: string, description: string, timeToRelease: string, status: string) {
        return this.http.post('/api/kanban/' + kanbanName + '/task', { name, assignee, description, timeToRelease, status });
    }

    deleteTask(kanbanName: string, taskName: string) {
        return this.http.delete('/api/kanban/' + kanbanName + '/task/' + taskName);
    }

    updateTask(kanbanName: string, oldName: string, name: string, assignee: string, description: string, timeToRelease: string, status: string) {
        return this.http.patch('/api/kanban/' + kanbanName + '/task/' + oldName, { name, assignee, description, timeToRelease, status });
    }
}
